rockcraft/lifecycle.py

```diff
@@ -18,17 +18,41 @@
 """Lifecycle integration."""
 
 import datetime
+import os
 import subprocess
 from pathlib import Path
 
 from craft_cli import EmitterMode, emit
 
+from rockcraft.errors import RockcraftInitError
+
 from . import oci, providers, utils
 from .parts import PartsLifecycle, Step
 from .project import Project, load_project
 from .providers import capture_logs_from_instance
 
 
+def init(rockcraft_yaml_content: str) -> None:
+    """Initialize a rockcraft project.
+
+    :param rockcraft_yaml_content: Content of the rockcraft.yaml file
+    :type rockcraft_yaml_content: str
+    :raises RockcraftInitError: raises initialization error in case of conflicts
+    with existing rockcraft.yaml files
+    """
+    rockcraft_yaml_path = "rockcraft.yaml"
+
+    if os.path.exists(rockcraft_yaml_path):
+        raise RockcraftInitError("{} already exists!".format(rockcraft_yaml_path))
+    elif os.path.exists(f".{rockcraft_yaml_path}"):
+        raise RockcraftInitError(f".{rockcraft_yaml_path} already exists!")
```

Comment:
This is interesting, I didn't know rockcraft supported `.rockcraft.yaml`?

This one is minor but I thought it amusing that you chose `.format()` in some places and f-strings in other places, heh

Reply:
this is basically the "power" of the copy&paste https://github.com/snapcore/snapcraft/blob/main/snapcraft_legacy/internal/lifecycle/_init.py#L51

:rofl: 

fixed it